src/tests/dbus-tests/test_loop.py

```diff
@@ -0,0 +1,99 @@
+#!/bin/python3
+
+import unittest
+import storagedtestcase
+import subprocess
+import time
+import dbus
+import glob
+import os
+
+class StoragedLoopDeviceTest(storagedtestcase.StoragedTestCase):
+    """This is SwapSpace related functions unit test"""
+
+    LOOP_DEVICE_FILENAME = 'loop_device.img'
+
+    def get_device(self, dev_name):
+        dev = self.get_object('', '/block_devices/' + os.path.basename(dev_name))
+        return dev
+    #enddef
+
+
+    def get_loop_iface(self, device):
+        iface = dbus.Interface(device, dbus_interface=self.iface_prefix + '.Loop')
+        return iface
+    #enddef
+
+
+    def setUp(self):
+        # create a file and fill it with some data
+
+        self.FNULL = open(os.devnull, 'w')
+
+        subprocess.call(['dd', 'if=/dev/zero', 'of=%s'%self.LOOP_DEVICE_FILENAME,
+                         'bs=10MiB', 'count=1'], stdout=self.FNULL, stderr=self.FNULL)
+        # create loop device (losetup will return device name or nothing)
+        with subprocess.Popen(['losetup', '--find', '--show', self.LOOP_DEVICE_FILENAME],
+                    stdout=subprocess.PIPE, bufsize=30, universal_newlines=True) as p:
+            for line in p.stdout:
+                self.dev_name = line
+            #endfor
+        #endwith
+        assert self.dev_name != ''
+        self.dev_name = self.dev_name.strip()
+        self.device = self.get_device(self.dev_name)
+        self.iface = self.get_loop_iface(self.device)
+    #enddef
+
+
+    def tearDown(self):
+        # tear down loop device
+        subprocess.call(['umount', self.dev_name], stderr=self.FNULL)
+        subprocess.call(['losetup', '--detach', self.dev_name], stderr=self.FNULL)
+        os.remove(self.LOOP_DEVICE_FILENAME)
+        self.FNULL.close()
+    #enddef
+
+
+    def test_10_Delete(self):
+        # check that loop device exists
+        with subprocess.Popen(['losetup', '--list'],
+                    stdout=subprocess.PIPE, bufsize=80, universal_newlines=True) as p:
+            for line in p.stdout:
+                if self.dev_name in line:
+                    break
+                #endif
+            else:
+                # for cycle did not exit with break
+                self.fail('Test loop device "%s" not found' %self.dev_name)
+            #endfor
+        #endwith
+
+        # remove loop device
+        self.iface.Delete(self.no_options)
+        subprocess.call(['udevadm', 'settle'])
+
+        # check that loop device does not exist anymore
+        with subprocess.Popen(['losetup', '--list'],
+                    stdout=subprocess.PIPE, bufsize=80, universal_newlines=True) as p:
+            for line in p.stdout:
+                if self.dev_name in line:
+                    self.fail('Test loop device was not deleted' %self.dev_name)
+                #endif
+            #endfor
+        #endwith
+    #enddef
+
```

Comment:
The device is kept on dbus, never removed. Documentation does not mention this. It seems to be an issue of udisks2/storaged. Issue is not addressed at the moment since fixing it would result in  different behavior of udisks2 and storaged. Added remainder comment to the test.